### PR TITLE
Implementation of the PCS power capping functionality

### PR DESCRIPTION
**Pull request by alvarez3-hpe**
### Summary and Scope

This implements the power capping domain functions and includes:
- Preliminary implementation of a credstore (vault) layer with credential cache
- Power capping domain functions for executing snapshot, patch, and task retrieval API operations
- Power capping storage functions

NOTE: This includes the HSM layer implementation in the [Implemented HSM layer for PCS](https://github.com/Cray-HPE/hms-power-control/pull/8) PR and should be merged AFTER that PR.

### Issues and Related PRs

* Resolves [CASMHMS-5341](https://jira-pro.its.hpecorp.net:8443/browse/CASMHMS-5341)
* Merge after [Implemented HSM layer for PCS](https://github.com/Cray-HPE/hms-power-control/pull/8)

### Testing

Tested on:

* locally
* unittest

### Risks and Mitigations

No risk. Not used anywhere.

### CODE REVIEW

File: internal/hsm/hsm_test.go [hunk 1]
```diff
@@ -0,0 +1,328 @@
+// MIT License
```
Comment (mpkelly-hpe) [inline internal/hsm/hsm_test.go:1]: This is going to cause a conflict, since my PR has this file already.

Comment (alvarez3-hpe) [reply internal/hsm/hsm_test.go:1]: I made note in my PR that yours should get merged 1st. The code in internal/hsm is an exact copy of the code from your PR as of this morning plus some additions I needed for power capping. The conflicts shouldn't be too bad.

Comment (alvarez3-hpe) [reply internal/hsm/hsm_test.go:1]: I mark this comment resolved once your PR has merged to develope and I've dealt with the conflicts

**Review by mjendrysik-hpe — APPROVED** [review]

File: internal/hsm/hsm_test.go (hunk 1 above)
Comment (alvarez3-hpe) [reply internal/hsm/hsm_test.go]: I've updated this PR with develop after Matt's HSM layer was merged

**Review by mpkelly-hpe — APPROVED** [review]

**Pull request merged**